pkg/util/conditions/conditions.go

```diff
@@ -24,3 +25,9 @@ func GetDeploymentCondition(deployment *appsv1.Deployment, conditionType appsv1.
 	}
 	return nil
 }
+
+func DeepCopyConditions(in []machinev1.Condition) []machinev1.Condition {
+	out := make([]machinev1.Condition, len(in))
+	copy(out, in)
```

Comment:
Is this actually deep copying? Does copy not just copy the structs across here and therefore any pointers within the `machinev1.Condition` would not actually be deep copied?

Expected more like

```
for _, cond := range in {
  out = append(out, *cond.DeepCopy())
}
```

Reply:
I think you're right RE dealing with pointers, but given we're passing by value and everything in machinev1.Condition is a string (and not a pointer) I think this is OK?

```go
// Condition defines an observation of a Machine API resource operational state.
type Condition struct {
	// Type of condition in CamelCase or in foo.example.com/CamelCase.
	// Many .condition.type values are consistent across resources like Available, but because arbitrary conditions
	// can be useful (see .node.status.conditions), the ability to deconflict is important.
	// +required
	// +kubebuilder:validation:Required
	Type ConditionType `json:"type"`

	// Status of the condition, one of True, False, Unknown.
	// +required
	Status corev1.ConditionStatus `json:"status"`

	// Severity provides an explicit classification of Reason code, so the users or machines can immediately
	// understand the current situation and act accordingly.
	// The Severity field MUST be set only when Status=False.
	// +optional
	Severity ConditionSeverity `json:"severity,omitempty"`

	// Last time the condition transitioned from one status to another.
	// This should be when the underlying condition changed. If that is not known, then using the time when
	// the API field changed is acceptable.
	// +required
	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`

	// The reason for the condition's last transition in CamelCase.
	// The specific API may choose whether or not this field is considered a guaranteed API.
	// This field may not be empty.
	// +optional
	Reason string `json:"reason,omitempty"`

	// A human readable message indicating details about the transition.
	// This field may be empty.
	// +optional
	Message string `json:"message,omitempty"`
}
```

Reply:
For reference, yes, I think you're right, but, with the approach you have, it becomes fragile if we were ever to add a pointer into this in the future, which could lead to a nasty bug, so I'd prefer we do it "properly".

For reference, the [original implementation](https://github.com/openshift/api/blob/764241ac5317cbbfeb2720195cca25d03e754c27/machine/v1beta1/zz_generated.deepcopy.go#L426-L446) does iterate as suggested